src/BloomExe/Book/BookStorage.cs

```diff
@@ -859,36 +859,57 @@ private void MakeCssLinksAppropriateForStoredFile(HtmlDom dom)
 		}
 
 
-		private string SanitizeNameForFileSystem(string name)
+		internal static string SanitizeNameForFileSystem(string name)
 		{
-			foreach(char c in PathUtilities.GetInvalidOSIndependentFileNameChars())
+			name = RemoveDangerousCharacters(name);
+			if (name.Length == 0)
 			{
-				name = name.Replace(c, ' ');
+				// The localized default book name could itself have dangerous characters.
+				name = RemoveDangerousCharacters(BookStarter.UntitledBookName);
+				if (name.Length == 0)
+					name = "Book";	// This should absolutely never be needed, but let's be paranoid.
```

Comment:
The final check and use of unadorned "Book" happens only if the translator used only invalid characters as the localization of "Book".  That's why is should never be needed, and why i refer to the check as paranoid.  The fallback to the localized UntitledBook is still there as the initial fallback.